site/frontend/src/storage.ts

```diff
@@ -0,0 +1,48 @@
+import {ref, Ref, watch} from "vue";
+
+/**
+ * Value that is persisted in LocalStorage.
+ */
+class StoredValue<T> {
+  private _value: T;
+
+  constructor(private key: string, defaultValue: T) {
+    const found = localStorage.getItem(key);
+    if (found === null) {
+      this._value = defaultValue;
+    } else {
+      this._value = JSON.parse(found);
+    }
+  }
+
+  get value(): T {
+    return this._value;
+  }
+
+  store(value: T) {
+    this._value = value;
+    console.debug(`Changing local preference ${this.key} to ${value}`);
+    localStorage.setItem(this.key, JSON.stringify(value));
```

Comment:
Then here and the `JSON.parse(found)` above, we'd also probably need to try/catch json encoding/decoding, and clear the k/v in case of errors. There could be bugs, related to weird edge cases, values like "" or undefined and so on, or people manually changing the values in the browser, that eagerly clearing the prefs could be some kind of protection.